org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java

```diff
@@ -13,50 +13,178 @@
+		URI targetUri = Paths.get(RESOURCES_DIR, folder.newFile("managementplan_exceptionConnection.zip").getAbsolutePath()).toUri();
+		BPMN4Tosca2BpelTest.class.getResource(".");
+		Bpmn4Tosca2Bpel transformer = new Bpmn4Tosca2Bpel();
+		transformer.transform(srcUri, targetUri);
+		fail();
+	}
+
+	@Test(expected = ParseException.class)
+	public void testTransformGatewayOneWayOut()
+			throws ParseException,
+			PlanWriterException,
+			URISyntaxException, IOException {
+		URI srcUri = Paths.get(RESOURCES_DIR, "bpmn4tosca.exclusivegatewayOneWayOut.json").toUri();
+		URI targetUri = Paths.get(RESOURCES_DIR, folder.newFile("managementplan.exclusivegateway.zip").getAbsolutePath()).toUri();
+		BPMN4Tosca2BpelTest.class.getResource(".");
+		Bpmn4Tosca2Bpel transformer = new Bpmn4Tosca2Bpel();
+		transformer.transform(srcUri, targetUri);
+		fail();
+
+	}
+
+	@Test(expected = ParseException.class)
+	public void testTransformExceptionTwoConnections() throws ParseException,
+			PlanWriterException,
+			URISyntaxException, IOException {
+		URI srcUri = Paths.get(RESOURCES_DIR, "bpmn2tosca_exceptionTwoConnections.json").toUri();
+		URI targetUri = Paths.get(RESOURCES_DIR, folder.newFile("managementplan_exceptionConnection.zip").getAbsolutePath()).toUri();
+		BPMN4Tosca2BpelTest.class.getResource(".");
+		Bpmn4Tosca2Bpel transformer = new Bpmn4Tosca2Bpel();
+		transformer.transform(srcUri, targetUri);
+		fail();
+	}
+
+	/**
+	 * @param archive1
+	 * @param archive2
+	 * @throws ZipException
+	 * @throws IOException
+	 */
+	public static final void assertZipEquals(String archive1, String archive2) throws ZipException, IOException  {
+		// Get Archives
+		ZipFile zipFile1 = new ZipFile(new File(archive1));
+		ZipFile zipFile2 = new ZipFile(new File(archive2));
+
+		// Get Member Hash
+		HashMap<String, ZipEntry> files1 = getMembers(zipFile1);
+		HashMap<String, ZipEntry> files2 = getMembers(zipFile2);
+
+		// Compare Files
+		assertMembersEqual(zipFile1, files1, zipFile2, files2);
+	}
+	/**
+	 * @param archive
+	 * @return
+	 * @throws IOException
+	 */
+	private static final HashMap<String, ZipEntry> getMembers(ZipFile archive) throws IOException {
+		HashMap<String, ZipEntry> map = new HashMap<String, ZipEntry>();
+		@SuppressWarnings("unchecked")
+		Enumeration<ZipEntry> entries = (Enumeration<ZipEntry>) archive.entries();
+		while (entries.hasMoreElements()) {
+			ZipEntry entry = entries.nextElement();
+			map.put(entry.getName(), entry);
+		}
+		return map;
+	}
+
+	/**
+	 * @param zip1
+	 * @param files1
+	 * @param zip2
+	 * @param files2
+	 * @throws IOException
+	 */
+	private static final void assertMembersEqual(ZipFile zip1, HashMap<String, ZipEntry> files1,
+												 ZipFile zip2, HashMap<String, ZipEntry> files2) throws IOException {
+		if (files1.size() != files2.size()) {
+			fail("Different Sizes, expected " + Integer.toString(files1.size()) + " found " + Integer.toString(files2.size()));
+		}
+
+		for (String key : files1.keySet()) {
```

Comment:
Try Java8 lambda:

    files1.keySet().stream().anyMatch(key -> !files2.containsKey(Key))

I think, tehre should be a way to compare collections. Why is `Assert.assertEquals(files1.keySet(), files2.keySet())` not possible?